aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -193,9 +193,10 @@ Tensor& eye_out_cpu(Tensor& result, int64_t n, int64_t m) {
   int64_t sz = std::min<int64_t>(n, m);
   AT_DISPATCH_ALL_TYPES(result.scalar_type(), "eye", [&]() -> void {
     scalar_t* result_data = result.data<scalar_t>();
-    for(int64_t i = 0; i < sz; i++) {
-      result_data[i*(result.strides()[0] + result.strides()[1])] = 1;
-    }
+    at::parallel_for(0, sz, 1, [&](int64_t p_begin, int64_t p_end) {
```

Comment:
are you sure a `grain_size` of 1 is the best here? I'd expect that for small tensors the overhead of parallelizing it would dominate.

Reply:
I changed this to `internal::GRAIN_SIZE`. Hope it is sufficient.